Base/res/js/Spreadsheet/runtime.js

```diff
@@ -351,6 +351,29 @@ function variance(cells) {
     return (count * squares - sums * sums) / count;
 }
 
+function mode(cells) {
+    const counts = numericReduce((map, x) => {
+        if (!map.has(x))
+            map.set(x, 0);
+        map.set(x, map.get(x) + 1);
+        return map;
+    },
```

Comment:
That's the sort of one-liner I'm always looking out for, though I wasn't creative enough here :^)

Reply:
That's not exactly equivalent is it? You'd be doing `map.set(x, 0+1)` instead of `map.set(x,0)`

Not sure if that makes a difference in this case though

Reply:
@mustafaquraish There's no difference. We set the value of x in the map to at least 1 in any case. The `if` is just defaulting the map to 0 (cf. python's `defaultdict`)

Reply:
Ah right, there's no `else` for the second line, silly me